loky/process_executor.py

```diff
@@ -172,6 +171,9 @@ def _python_exit():
         thread.join()
 
 
+# Module variable to register the at_exit call
+process_pool_executor_at_exit = None
```

Comment:
How is this related to the potential race condition / deadlock at shutdown?

Reply:
This is not, I fixed it here as I realized the issue while fixing the logging behavior to get proper debugging info.